ipam/client/backends/phpipam.py

```diff
@@ -258,6 +258,38 @@ def delete_ip(self, ipaddress):
         self.db.commit()
         return True
 
+    def delete_subnet(self, subnet, empty_subnet=False):
+        """Delete a subnet in IPAM. subnet must be an
+        instance of ip_network with correct prefix length.
+        If empty_subnet is True, we will remove all IP addresses
+        in the subnet. Otherwise, we will raise an exception.
+        """
+        subnet_id = self.find_subnet_id(subnet)
+        if subnet_id is None:
+            raise ValueError("Unable to get subnet id from database"
+                             "for subnet %s/%s"
+                             % (subnet.network_address,
+                                subnet.prefixlen))
+        self.cur.execute("SELECT ip_addr FROM ipaddresses \
```

Comment:
There is a method named get_allocated_ips_by_subnet_id, you should use it to prevent code duplication.
Also, it will give you the list of IPs to remove if you need to suppres them later.